packages/@lwc/integration-karma/test/spread/index.spec.js

```diff
@@ -0,0 +1,43 @@
+import { createElement } from 'lwc';
+import Test from 'x/test';
+
+describe('lwc:spread', () => {
+    let elm, simpleChild, overriddenChild;
+    beforeEach(() => {
+        elm = createElement('x-test', { is: Test });
+        document.body.appendChild(elm);
+        simpleChild = elm.shadowRoot.querySelector('.x-child-simple');
+        overriddenChild = elm.shadowRoot.querySelector('.x-child-overridden');
+        spyOn(console, 'log');
+    });
+    it('should render basic test', () => {
+        expect(simpleChild.shadowRoot.querySelector('span').textContent).toEqual('Name: LWC');
+    });
+    it('should assign onclick', () => {
+        simpleChild.click();
+        // eslint-disable-next-line no-console
+        expect(console.log).toHaveBeenCalledWith('spread click called', simpleChild);
+    });
+    it('should override values in template', () => {
+        expect(overriddenChild.shadowRoot.querySelector('span').textContent).toEqual('Name: Aura');
+    });
+    it('should assign onclick along with the one in template', () => {
+        overriddenChild.click();
+        // eslint-disable-next-line no-console
+        expect(console.log).toHaveBeenCalledWith('spread click called', overriddenChild);
+        // eslint-disable-next-line no-console
+        expect(console.log).toHaveBeenCalledWith(
+            'template click called',
+            jasmine.any(Object) /* component */
+        );
+    });
+
+    it('should assign props to standard elements', () => {
+        expect(elm.shadowRoot.querySelector('span').className).toEqual('spanclass');
+    });
+    it('should assign props to dynamic elements', () => {
+        expect(
+            elm.shadowRoot.querySelector('x-cmp').shadowRoot.querySelector('span').textContent
+        ).toEqual('Name: Dynamic');
+    });
```

Comment:
Some more test cases that may be worth testing:

- The key inside of the `spanProps` is changed
- The `className` key in `spanProps` is deleted
- The value of `className` in `spanProps` is set to `undefined`
- The entire `spanProps` object changes to a new object like `{ className: 'foo' }`
- The entire `spanProps` object changes to an empty object

Reply:
I completely missed the reactivity part of it. I actually found a few issues with reactivity after adding these tests.